federation/pkg/kubefed/unjoin.go

```diff
@@ -81,6 +81,16 @@ func unjoinFederation(f cmdutil.Factory, cmdOut, cmdErr io.Writer, config util.A
 		return nil
 	}
 
+	//we need to ensure deleting the config map created in the deregistered cluster
+	//this configmap was created when the cluster joined this federation to aid
+	//the kube-dns of that cluser to aid service discovery
+	unjoinClusterFactory := config.HostFactory(unjoinFlags.Name, unjoinFlags.Kubeconfig)
+	err = deleteConfigMap(unjoinClusterFactory)
```

Comment:
I don't think we should just delete it. We need to be safer than that. We need to remove the value for this federation from ConfigMap's `federations` key and if it is empty, we should delete the ConfigMap.